src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+                system.assertNotEquals(false, addr.Default_Address__c);
+            }
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Update Contact's Mailing address with just a case change, to cause an update to default Address object  
+    verify:
+        existing Default Address updated
+        Contact has updated Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrExistingCaseOnly() { 
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // modify some of the Contact addresses directly
+        // NOTE: we only modify 1 field, so it will be treated as an update to an existing address!
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = 'Direct Street Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' direct street  edit ';  // whitespace and casing should not count as a real edit.
+            contactsAddrs.contacts[i].MailingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' DIRECT STREET  EDIT ';  // only change casing.
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.equals('DIRECT STREET  EDIT'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.